compat/unistd.cc

```diff
@@ -65,11 +81,9 @@ xopen(const char *filename, int oflag, int pmode)
 int
 xread(int fd, void * buf, size_t sz)
 {
-    char l_so_type[sizeof(int)];
-    int l_so_type_siz = sizeof(l_so_type);
-    SOCKET sock = _get_osfhandle(fd);
+    const auto sock = _get_osfhandle(fd);
 
```

Comment:
Missing INVALID_HANDLE_VALUE code snippet. Just like in other cases where that snippet is present, we do not want to call _read() with an invalid handle here; isSocket() call addition does not change that overall logic/motivation.

Same for xwrite(), but xclose() got it right.

Reply:
Changed it for cleanliness and efficiency, however I the error case would still be handled correctly that I can see:
if the fd is invalid, then sock is INVALID_HANDLE_VALUE (-1); isSocket will return false, falling through to the _write and _read calls, which honour the right semantics (return -1 and set errno to EBADF)